examples/fetch.rs

```diff
@@ -0,0 +1,175 @@
+// `error_chain!` can recurse deeply
+#![recursion_limit = "1024"]
+
+#[macro_use]
+extern crate error_chain;
+extern crate futures;
+extern crate hyper;
+extern crate hyper_tls;
+extern crate native_tls;
+extern crate serde_json;
+extern crate tokio_core;
+
+use std::io::{self, Write};
+use std::process;
+
+use futures::{future, Future, Stream};
+use hyper::Client;
+use hyper_tls::HttpsConnector;
+use serde_json::Value;
+use serde_json::map::Map;
+use tokio_core::reactor::Core;
+
+// We'll put our errors in a `errors` module
+mod errors {
+    // Create the Error, ErrorKind, ResultExt, and Result types
+    #[allow(unused_doc_comment)]
+    error_chain!{
+        // Automatic conversions between this error chain and other
+        // error types not defined by the `error_chain!`.
+        foreign_links {
+            IO(::std::io::Error);
+            TLS(::native_tls::Error);
+            URI(::hyper::error::UriError);
+            Hyper(::hyper::error::Error);
+            JSON(::serde_json::Error);
+        }
+    }
+}
+
+fn download<F>(core: &mut Core, uri: &str, mut action: F, debug: bool) -> errors::Result<()>
+    where F: FnMut(Map<String, Value>) -> errors::Result<()> {
+    let handle = core.handle();
+    let c = Client::configure().connector(HttpsConnector::new(4, &handle)?).build(&handle);
+    let future = c.get(uri.parse()?).and_then(|res| {
+        future::result(if debug {
+            writeln!(io::stderr(), "Response: {}", res.status())
+        } else { Ok(()) }.map_err(|e| e.into())).and_then(|_| {
+            res.body().concat2()})});
+    let res = core.run(future)?;
+
+    let json: Value = serde_json::from_slice(&res)?;
+    let json = if let Value::Object(m) = json {
+        m
+    } else {
+        return Err(io::Error::new(io::ErrorKind::Other, format!("Malformed JSON: {:?}", json)).into())
+    };
+
+    if debug {
+        writeln!(io::stderr(), "#json == {}", json.len())?;
+        writeln!(io::stderr(), "License list version {}", json["licenseListVersion"])?;
+    }
+    action(json)
+}
+
+fn main1() -> errors::Result<()> {
+    let debug = std::env::args().any(|e| e == "-d");
+
+    let mut stdout = io::stdout();
+    let mut stderr = io::stderr();
+
+    writeln!(stdout, "\
+/*
+ * whitelist fetched from http://spdx.org/licenses/
+ *
+ * AUTO-GENERATED BY examples/fetch.rs
+ * DO NOT MODIFY
+ *
+ * cargo run --example fetch-license-list-from-spdx > spdx.rs
+ */
+")?;
+
+    let mut core = Core::new()?;
+
+    download(&mut core, "https://raw.githubusercontent.com/spdx/license-list-data/master/json/licenses.json", |json| {
```

Comment:
Instead of floating with master, I think it's better to take a tag argument and use that here.  For example:

```console
cargo run --example fetch-license-list-from-spdx v3.0
```

would pull from https://raw.githubusercontent.com/spdx/license-list-data/v3.0/json/licenses.json

Reply:
Git tag is now supported (sorry for the delay!)